sufia-models/app/services/sufia/generic_file_indexing_service.rb

```diff
@@ -9,7 +9,18 @@ def generate_solr_document
         solr_doc[Solrizer.solr_name('file_format', :facetable)] = object.file_format
         solr_doc['all_text_timv'] = object.full_text.content
         solr_doc[Solrizer.solr_name('file_size', STORED_INTEGER)] = object.content.size.to_i
+        # Index the Fedora-generated SHA1 digest to create a linkage
+        # between files on disk (in fcrepo.binary-store-path) and objects
+        # in the repository.
+        solr_doc[Solrizer.solr_name('digest', :symbol)] = digest_from_content
       end
     end
+
+    private
+
+      def digest_from_content
+        return unless object.content.has_content?
+        object.content.metadata.ldp_source.graph.query(predicate: RDF::Vocab::Fcrepo4.digest).first.object.to_s
```

Comment:
Actually, isn't it already there? https://github.com/projecthydra/active_fedora/blob/d6784e23a7ad324e2d03a23c7c063fc888b89398/lib/active_fedora/file/attributes.rb#L17